app/containers/LoginPage/components/TermsNotAccepted.tsx

```diff
@@ -0,0 +1,115 @@
+import React, { useMemo } from 'react';
+import { Formik } from 'formik';
+import { useIntl } from 'react-intl';
+import { useDispatch } from 'react-redux';
+
+import { RoutePath } from 'global/constants';
+import { termsAcceptRequest, termsAcceptSaga } from 'global/reducers/auth';
+
+import FormikInput from 'components/FormikInput';
+import LinkButton from 'components/Button/LinkButton';
+import Button from 'components/Button';
+import Row from 'components/Row';
+import TermsCheckbox from 'components/TermsCheckbox';
+import ErrorAlert from 'components/ErrorAlert';
+import FormikForm from 'components/FormikForm';
+
+import { useInjectSaga } from 'redux-injectors';
+import { generateTermsValidationSchema } from '../utils';
+import messages from '../messages';
+
+type Props = {
+  goBack: () => void;
+  error: string;
+  loading: boolean;
+  termsExtraFields: {
+    email: string;
+    firstName: Nullable<string>;
+    lastName: Nullable<string>;
+  };
+};
+
+const TermsNotAccepted = ({
+  error,
+  goBack,
+  termsExtraFields,
+  loading,
+}: Props) => {
+  const { formatMessage } = useIntl();
+  const dispatch = useDispatch();
+
+  useInjectSaga({ key: 'termsAcceptSaga', saga: termsAcceptSaga });
+  const validationSchema = useMemo(
+    () => generateTermsValidationSchema(formatMessage),
+    [],
+  );
+
+  const onSubmit = (values: any) => {
+    dispatch(termsAcceptRequest(values, goBack));
+  };
+
+  return (
+    <Formik
+      validationSchema={validationSchema}
+      initialValues={termsExtraFields}
+      onSubmit={onSubmit}
+    >
+      {({ handleSubmit }) => {
+        const sharedProps = {
+          inputProps: {
+            height: 46,
+            width: '100%',
+          },
+          mb: 20,
+        };
+
+        return (
+          <FormikForm>
+            <Row width="100%">
+              <FormikInput
+                formikKey="firstName"
+                placeholder={formatMessage(messages.firstNameLabel)}
+                label={formatMessage(messages.firstNameLabel)}
+                type="text"
+                mr={20}
```

Comment:
I suggest adding `gap={20}` to the parent Row component instead of adding margin to child elements to take an advantage from flex container